src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/StorageWatcher.cs

```diff
@@ -69,14 +120,82 @@ public virtual async Task CreateOrUpdatePackage(Package pkg)
+                cacheLastModifiedTime = DateTime.UtcNow;
+            } while (true);
         }
 
+       
+        private async Task StartBlobDownload(bool startup = false)
+        {
+            var detectorsList = await storageService.GetEntitiesByPartitionkey("Detector");
+            var gists = await storageService.GetEntitiesByPartitionkey("Gist");
+            var entitiesToLoad = new List<DiagEntity>();
+            var filteredDetectors = LoadOnlyPublicDetectors ? detectorsList.Where(row => !row.IsInternal).ToList() : detectorsList.Where(row => row.IsInternal);
+            if(!startup)
+            {
+                entitiesToLoad.AddRange(filteredDetectors.Where(s => s.Timestamp >= cacheLastModifiedTime).ToList());
+                entitiesToLoad.AddRange(gists.Where(s => s.Timestamp >= cacheLastModifiedTime).ToList());
+            } else
+            {
+                entitiesToLoad.AddRange(filteredDetectors.ToList());
+                entitiesToLoad.AddRange(gists);
+            }
+
+            try
+            {
+                foreach (var entity in entitiesToLoad)
+                {
+                    var assemblyData = await storageService.GetBlobByName($"{entity.RowKey.ToLower()}/{entity.RowKey.ToLower()}.dll");
+                    if (assemblyData == null || assemblyData.Length == 0)
+                    {
+                        continue;
+                    }
+
+                    // initializing Entry Point of Invoker using assembly
+                    Assembly temp = Assembly.Load(assemblyData);
+                    EntityType entityType = EntityType.Signal;
+                    if(entity.PartitionKey.Equals("Gist"))
+                    {
+                        entityType = EntityType.Gist;
+                    }
+                    else if (!Enum.TryParse<EntityType>(entity.DetectorType, out entityType))
+                    {
+                        entityType = EntityType.Signal;
+                    }
+                    EntityMetadata metaData = new EntityMetadata(string.Empty, entityType);
+                    var newInvoker = new EntityInvoker(metaData);
+                    newInvoker.InitializeEntryPoint(temp);
+
+                    if (_invokerDictionary.TryGetValue(entityType, out ICache<string, EntityInvoker> cache) && newInvoker.EntryPointDefinitionAttribute != null)
+                    {
+                        DiagnosticsETWProvider.Instance.LogAzureStorageMessage(nameof(StorageWatcher), $"Updating cache with new invoker with id : {newInvoker.EntryPointDefinitionAttribute.Id}");
+                        cache.AddOrUpdate(newInvoker.EntryPointDefinitionAttribute.Id, newInvoker);
+                    }
+                    else
+                    {
+                        DiagnosticsETWProvider.Instance.LogAzureStorageMessage(nameof(StorageWatcher), $"No invoker cache exist for {entityType}");
```

Comment:
so we dont need to initialize the cache for a entity type here (likf for Gist)?
We assume that the cache has to be there for supported entity types?

If yes, then can you make this message as warning as this is not desired path

Reply:
In the constructor we initialize the cache for Gist & Detector. I can change it to warning